astropy/cosmology/_io/cosmology.py

```diff
@@ -13,24 +13,34 @@
     True
 """
 
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
 from astropy.cosmology.connect import convert_registry
 from astropy.cosmology.core import _COSMOLOGY_CLASSES, Cosmology
 
-__all__ = []  # nothing is publicly scoped
+if TYPE_CHECKING:
+    from astropy.cosmology._typing import _CosmoT
+
+__all__: list[str] = []  # nothing is publicly scoped
 
 
-def from_cosmology(cosmo, /, **kwargs):
+def from_cosmology(
+    cosmo: _CosmoT, /, cosmology: type[_CosmoT] | str | None = None, **kwargs: object
+) -> _CosmoT:
     """Return the |Cosmology| unchanged.
 
     Parameters
     ----------
     cosmo : `~astropy.cosmology.Cosmology`
         The cosmology to return.
+    cosmology : `~astropy.cosmology.Cosmology` | str | None, optional
+        The |Cosmology| class to check against. If not `None`, ``cosmo`` is checked
```

Comment:
class or instance?

Reply:
Class

Reply:
The type annotation is definitely clearer than the docs. Unfortunate naming by me not to call the parameter `cosmo_cls`.

Reply:
We should try to keep type annotation pull requests separate from documentation updates, but in this case the docstring is actively lying, so it should be changed.